pycalphad/io/tdb.py

```diff
@@ -271,37 +271,55 @@ def _tdb_grammar(): #pylint: disable=R0914
 
 def _process_typedef(targetdb, typechar, line):
     """
-    Process the TYPE_DEFINITION command.
+    Process a TYPE_DEFINITION command.
+
+    Assumes all phases are entered into the database already and that the
+    database defines _typechar_map, which defines a map of typechar to the
+    phases that use it. Any phases that in the typechar dict for this will have
+    the model_hints updated based on this type definition, regardless of which
+    phase names may be defined in this TYPE_DEF line.
+
     """
+    matching_phases = targetdb._typechar_map[typechar]
+    del targetdb._typechar_map[typechar]
     # GES A_P_D BCC_A2 MAGNETIC  -1    0.4
     tokens = line.replace(',', '').split()
     if len(tokens) < 4:
         return
     keyword = expand_keyword(['DISORDERED_PART', 'MAGNETIC'], tokens[3].upper())[0]
     if len(keyword) == 0:
-        raise ValueError('Unknown keyword: {}'.format(tokens[3]))
+        raise ValueError('Unknown type definition keyword: {}'.format(tokens[3]))
+    if len(matching_phases) == 0:
+        warnings.warn(f"The type definition character `{typechar}` in `TYPE_DEFINITION {typechar} {line}` is not used by any phase.")
     if keyword == 'MAGNETIC':
-        # magnetic model (IHJ model assumed by default)
-        targetdb.tdbtypedefs[typechar] = {
-            'ihj_magnetic':[float(tokens[4]), float(tokens[5])]
+        # Magnetic model (IHJ model assumed by default)
```

Comment:
IHJ is not assumed here (the comment predates the implementation of the Xiong model). The `ihj_` model hints are a bit of a misnomer, since we use `ihj_magnetic_afm_factor == 0` as a check for turning the Xiong model on in `Model`. I think removing the comment is ok; possible renaming of the magnetic model hints is out of scope for this PR.

Reply:
I changed the existing comment to clarify that the `ihj_magnetic_` prefix is legacy and it's used for both Xiong and IHJ models.